packages/modules/settings/src/services/settings-module-service.ts

```diff
@@ -0,0 +1,388 @@
+  implements SettingsTypes.ISettingsModuleService
+{
+  protected baseRepository_: DAL.RepositoryService
+  protected readonly viewConfigurationService_: ModulesSdkTypes.IMedusaInternalService<
+    InferEntityType<typeof ViewConfiguration>
+  >
+  protected readonly userPreferenceService_: ModulesSdkTypes.IMedusaInternalService<
+    InferEntityType<typeof UserPreference>
+  >
+
+  constructor(
+    {
+      baseRepository,
+      viewConfigurationService,
+      userPreferenceService,
+    }: InjectedDependencies,
+    protected readonly moduleDeclaration: InternalModuleDeclaration
+  ) {
+    super(...arguments)
+    this.baseRepository_ = baseRepository
+    this.viewConfigurationService_ = viewConfigurationService
+    this.userPreferenceService_ = userPreferenceService
+  }
+
+  @InjectManager()
+  // @ts-expect-error
+  async retrieveViewConfiguration(
+    id: string,
+    config?: FindConfig<InferEntityType<typeof ViewConfiguration>>,
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<SettingsTypes.ViewConfigurationDTO> {
+    const viewConfig = await this.viewConfigurationService_.retrieve(
+      id,
+      config,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<SettingsTypes.ViewConfigurationDTO>(
+      viewConfig,
+      { populate: true }
+    )
+  }
+
+  @InjectTransactionManager()
+  // @ts-expect-error
+  async createViewConfigurations(
+    data:
+      | SettingsTypes.CreateViewConfigurationDTO
+      | SettingsTypes.CreateViewConfigurationDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<
+    SettingsTypes.ViewConfigurationDTO | SettingsTypes.ViewConfigurationDTO[]
+  > {
+    const input = Array.isArray(data) ? data : [data]
+
+    // Validate system defaults
+    for (const config of input) {
+      if (config.is_system_default && config.user_id) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          "System default view configurations cannot have a user_id"
+        )
+      }
+
+      if (config.is_system_default) {
+        // Check if a system default already exists for this entity
+        const existingDefault = await this.viewConfigurationService_.list(
+          {
+            entity: config.entity,
+            is_system_default: true,
+          },
+          { select: ["id"] },
+          sharedContext
+        )
+
+        if (existingDefault.length > 0) {
+          throw new MedusaError(
+            MedusaError.Types.DUPLICATE_ERROR,
+            `A system default view configuration already exists for entity: ${config.entity}`
+          )
+        }
+      }
+    }
+
+    const created = await this.viewConfigurationService_.create(
+      input,
+      sharedContext
+    )
+
+    const serialized = await this.baseRepository_.serialize<
+      SettingsTypes.ViewConfigurationDTO[]
+    >(created, { populate: true })
+
+    return Array.isArray(data) ? serialized : serialized[0]
```

Comment:
```suggestion  
  return await super.createViewConfigurations(input, sharedContext)
```
We can call the auto generated method that will also emit events